src/Controller/Api/FileManager.php

```diff
@@ -323,70 +326,35 @@ public function downloadFile(File $file, Datastore $datastore): Response
      *
      * @return Response
      */
-    public function downloadZipAllFiles(DatasetSubmission $datasetSubmission): Response
+    public function downloadZipAllFiles(DatasetSubmission $datasetSubmission, LogActionItemEventDispatcher $logActionItemEventDispatcher, Request $request): Response
     {
-        if (
-            $datasetSubmission->getDataset()->getAvailabilityStatus() !==
-            DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE
-            and
-            !$this->isGranted('CAN_EDIT', $datasetSubmission)
-        ) {
-            throw new AccessDeniedHttpException('File unavailable for download');
-        }
-
-        $zipFilePath = $this->getZipFilePath($datasetSubmission);
-        if ($zipFilePath) {
-            $response = new StreamedResponse(function () use ($zipFilePath) {
-                $outputStream = fopen('php://output', 'wb');
-                $fileStream = fopen($zipFilePath, 'r');
-                stream_copy_to_stream($fileStream, $outputStream);
-            });
-            $disposition = HeaderUtils::makeDisposition(
-                HeaderUtils::DISPOSITION_ATTACHMENT,
-                basename($zipFilePath)
+        $dataset = $datasetSubmission->getDataset();
+        $udi = $dataset->getUdi();
+        // Only log if this is downloaded from dataland. We don't log review downloads.
+        if ($request->headers->get('referer') and preg_match("/^.*\/data\/$udi$/", $request->headers->get('referer'))) {
+            $currentUser = $this->getUser();
+            if ($currentUser instanceof Account) {
+                $type = 'GoMRI';
+                $typeId = $currentUser->getUserId();
+            } else {
+                $type = 'Non-GoMRI';
+                $typeId = 'anonymous';
+            }
+            $logActionItemEventDispatcher->dispatch(
+                array(
+                    'actionName' => 'File Download',
+                    'subjectEntityName' => 'Pelagos\Entity\Dataset',
+                    'subjectEntityId' => $dataset->getId(),
+                    'payLoad' => array('userType' => $type, 'userId' => $typeId),
+                ),
+                'file_download'
             );
-            $response->headers->set('Content-Disposition', $disposition);
-            return $response;
-        } else {
-            throw new BadRequestHttpException('No Zip file found');
         }
-    }
 
-    /**
```

Comment:
Don't remove these functions, it's not part of this ticket. And they are still being used by other actions!